pkg/agent/run.go

```diff
@@ -74,7 +74,16 @@ func Run(cmd *cobra.Command, args []string) {
 		logs.Log.Fatalf("While evaluating configuration: %v", err)
 	}
 
-	go func() {
+	group, gctx := errgroup.WithContext(ctx)
+	defer func() {
+		// TODO: replace Fatalf log calls with Errorf and return the error
+		cancel()
+		if err := group.Wait(); err != nil {
+			logs.Log.Fatalf("failed to wait for controller-runtime component to stop: %v", err)
+		}
+	}()
```

Comment:
This code was cut and pasted earlier in the file so that the errgroup can be initialized and used for the management server (`/metrics`, `/healthz`, `/debug/pprof`) and for the VenafiConnection manager; both of which were previously run in unsupervised goroutines.